src/Components/CommentComposer.jsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from "react";
+import { postComment } from "../__utils__/api";
+import { useContext } from "react";
+import { UserContext } from "../Contexts/Contexts";
+
+export default function CommentComposer({ articleId, comments, setComments }) {
+
+    const [comment, setComment] = useState('')
+    const [formValid, setFormValid] = useState(false)
+    const [submitAttempted, setSubmitAttempted] = useState(false)
+    const [commentLoading, setCommentLoading] = useState(true)
+    const [error, setError] = useState(Error())
+    const user = useContext(UserContext)
+
+    useEffect(() => {
+        
+    }, [formValid, submitAttempted])
+
+    const validateForm = () => {
+        if (comment.trim() === '') {
+            setFormValid(false)
+            return false
+        }
+        setFormValid(true)
+        return true
+    }
+
+    const handleChange = (event) => {
+        setComment(event.target.value)
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        setCommentLoading(true)
+        setSubmitAttempted(true)
+        if (!validateForm()) {
+            setCommentLoading(false)
+            return
+        }
+        
+        setComments([{
+            comment_id: 'temporaryKey',
```

Comment:
This isnt very unique so if a user creates two comments then they would both have this exact same comment_id which isnt ideal.

Reply:
Thanks MKD! Fixed in https://github.com/liampjh34/nc-news/pull/9 ✅